lib.php

```diff
@@ -494,54 +494,78 @@ function block_coursefeedback_get_feedbackname($feedbackid = null) {
 
     return htmlentities($name);
 }
-
 /**
+ * This function gets the amount of votes for each answeroption (1-6) and each question.
+ * It calculates the amount of counted choices, the amount of abstentions and the average rating for each question
+ *
  * @param int $courseid
  * @param string $sort
- * @return array - 2-dimensional array of answers, ordered by question id
+ * @return array - 2-dimensional array of answers, ordered by question id as follows:
+ * [ "<questionid>" =>
+ *      [ "1" => <answercount>, ..., "6" => <answercount>, "average" =>..., "choicessum" =>..., "abstentions" => ... ]]
+ * @throws moodle_exception
  */
-function block_coursefeedback_get_answers($course, $feedbackid, $sort = "questionid") {
-    global $DB, $CFG, $USER;
+function block_coursefeedback_get_qanswercounts($course, $feedbackid) {
+    global $DB;
     $config = get_config("block_coursefeedback");
     $answers = array();
     $course = clean_param($course, PARAM_INT);
 
     if ($course <= 0) {
         throw new moodle_exception("invalidcourseid");
     }
-
-    $questions = block_coursefeedback_get_questions($feedbackid, $config->default_language);
+    // Get all the questions of the feedback
+    $questions = block_coursefeedback_get_questions_by_language($feedbackid, [current_language()]);
     $params = array("fid" => $feedbackid, "course" => $course);
     if (!empty($questions)) {
-        $count = count($questions);
-        foreach (array_keys($questions) as $question) {
-            $params["qid"] = $question;
+        // For each question and each answerpossibility, count the amount of the given answers
+        foreach ($questions as $question) {
+            $choicessum = 0;
+            $avsum = 0;
+            $questionid = $question->questionid;
+            $params["qid"] = $questionid;
             $sql = "SELECT answer,COUNT(*) AS count
-                      FROM {block_coursefeedback_answers}
-                     WHERE coursefeedbackid = :fid 
-                           AND questionid = :qid 
-                           AND course = :course
-                  GROUP BY answer";
-
+                    FROM {block_coursefeedback_answers}
+                    WHERE coursefeedbackid = :fid 
+                        AND questionid = :qid 
+                        AND course = :course
+                    GROUP BY answer";
```

Comment:
Hier war die Einrückung schon korrekt siehe [hier](https://moodledev.io/general/development/policies/codingstyle/sql#indentation).

Reply:
Resolved in 732883e39f13a2210e7df8f303e7d899c96a1fa7